Tests/Realm.Tests/Database/PrimaryKeyTests.cs

```diff
@@ -93,6 +124,91 @@ public void FailToFindByPrimaryKeyDynamicTests(Type type, object primaryKeyValue
             Assert.That(foundObj, Is.Null);
         }
 
+        [TestCaseSource(nameof(UpdatePKTestCases))]
+        public void UpdatePrimaryKey_DynamicTests(Type type, object firstValue, object secondValue)
+        {
+            var obj = (RealmObject)Activator.CreateInstance(type);
+            var pkProperty = type.GetProperties().Single(p => p.GetCustomAttribute<PrimaryKeyAttribute>() != null);
+            pkProperty.SetValue(obj, firstValue);
+
+            _realm.Write(() => _realm.Add(obj));
+
+            // Setting the PK to the original value should be a no-op
+            Assert.DoesNotThrow(() =>
+            {
+                _realm.Write(() =>
+                {
+                    obj.DynamicApi.Set(obj.ObjectSchema.PrimaryKeyProperty.Value.Name, Operator.Convert<RealmValue>(firstValue));
+                });
+            });
+
+            var ex = Assert.Throws<InvalidOperationException>(() =>
+            {
+                _realm.Write(() =>
+                {
+                    obj.DynamicApi.Set(obj.ObjectSchema.PrimaryKeyProperty.Value.Name, Operator.Convert<RealmValue>(secondValue));
+                });
+            });
+
+            Assert.That(ex.Message, Does.Contain("Once set, primary key properties may not be modified."));
+            Assert.That(ex.Message, Does.Contain(Operator.Convert<RealmValue>(firstValue).ToString()));
+            Assert.That(ex.Message, Does.Contain(Operator.Convert<RealmValue>(secondValue).ToString()));
+
+            if (TestHelpers.IsUnity)
+            {
+                return;
```

Comment:
Would it be helpful the clarify why the rest is not applicable / needs to be skipped for Unity?

Reply:
Generally anything that uses the dynamic keyword is not supported on Unity.

Reply:
👍